scene/3d/light.cpp

```diff
@@ -465,6 +470,20 @@ OmniLight::OmniLight() :
 	set_shadow_detail(SHADOW_DETAIL_HORIZONTAL);
 }
 
+String SpotLight::get_configuration_warning() const {
+	String warning = Spatial::get_configuration_warning();
```

Comment:
I think you should get the warning of the most direct ancestor, so here `Light`.